src/state.py

```diff
@@ -135,6 +144,10 @@ def from_charm(
             InvalidStateError: if the state is invalid.
         """
         try:
+            secret_id = indexer_relation_data.get("secret-user")
+            secret_content = charm.model.get_secret(id=secret_id).get_content()
```

Comment:
if secret not found?

Reply:
I validation error is raised. This is within a try-except block